src/linearization.jl

```diff
@@ -22,7 +23,45 @@ Sparse matrix `A`.
 
 See references [1] and [2] of CARLIN.md.
 """
-function build_matrix(F₁, F₂, N)
+function build_matrix(F₁, F₂, N; compress=false)
+    F₁ = sparse(F₁)
+    F₂ = sparse(F₂)
+    if compress
+        n = size(F₁)[1]
```

Comment:
for readability i suggest to refactor the whole code inside the `compress` into its own `build_matrix_compressed(F1, F2, N)` function (which is in L118 in this file)